src/trodes_to_nwb/spike_gadgets_raw_io.py

```diff
@@ -123,7 +132,19 @@ def _produce_ephys_channel_ids(
                     for i in range(int(n_total_channels / n_channels_per_chip))
                 ]
             )
-        return [item for sublist in x for item in sublist]
+
+        channel_names = [item for sublist in x for item in sublist]
+
+        if n_total_channels == n_channels_recorded:
+            # case where all channels are recorded, no censoring required
+            return channel_names
+
+        if hw_channels_recorded is None:
```

Comment:
It would be good to add a test for this. I would actually be more specific and use something like
```suggestion
        if not hw_channels_recorded or len(hw_channels_recorded) != n_channels_recorded:
```